services/rule_engine/src/main/java/agents/Smith.java

```diff
@@ -1,27 +1,77 @@
 package agents;
 
 import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
 
+import com.mindsmiths.gsheetsAdapter.GSheetsAdapterAPI;
 import com.mindsmiths.gsheetsAdapter.reply.Spreadsheet;
 import com.mindsmiths.ruleEngine.model.Agent;
+import com.mindsmiths.ruleEngine.util.Agents;
 import com.mindsmiths.ruleEngine.util.Log;
 
+
 import lombok.AllArgsConstructor;
 import lombok.Data;
 import lombok.ToString;
 
+
 @Data
 @ToString(callSuper = true)
 @AllArgsConstructor
 public class Smith extends Agent {
 
     LocalDateTime lastTableUpdate = null;
+    int nextMailRow = 0;
+    int nextAnsRow = 0;
+    Spreadsheet sheet;
 
+    // Agent id
     public Smith() {
         this.id = "SMITH";
     }
 
-    public void processSheet(Spreadsheet spreadsheet){
-        Log.info(spreadsheet);
+    // Getting the sheet as an object and getting the first free row in the spreadsheet
+    public void processSpreadsheet(Spreadsheet spreadsheet){
+        sheet = spreadsheet;
+        List<Map<String, String>> emails = spreadsheet.getSheets().get("Mailovi");
+        List<Map<String, String>> answers = spreadsheet.getSheets().get("Odgovori");
+        nextMailRow = emails.size() + 2;
```

Comment:
Why +2?

Reply:
Because we want the next free row which is normally +1 but because of the category titles it must be +2